_playwright-tests/tests/API/ListTemplatesWithFilters.spec.ts

```diff
@@ -0,0 +1,173 @@
+            });
+            expect(templates_arch_search.data).toHaveLength(2);
+        });
+
+        await test.step('List templates by version', async () => {
+            const templates_version_search = await new TemplatesApi(client).listTemplates({
+                version: '10',
+            });
+            expect(templates_version_search.data).toHaveLength(1);
+        });
+
+        await test.step('List templates by architecture, version, limit, offset, sortby', async () => {
+            const templates_mixed_search = await new TemplatesApi(client).listTemplates({
+                arch: 'aarch64',
+                version: '9',
+                limit: 1,
+                offset: 0,
+                sortBy: 'name',
+            });
+
+            const expectedTemplate = createdTemplates
+                .filter(t => t.arch === 'aarch64' && t.version === '9')
+                .sort((a, b) => a.name!.localeCompare(b.name!))[0];
+
+            expect(templates_mixed_search.data).toHaveLength(1);
+            expect(templates_mixed_search.data?.[0]?.name).toBe(expectedTemplate.name);
+        });
+
+        await test.step('List templates by repository UUIDs', async () => {
+            const templates_repository_uuids_search = await new TemplatesApi(client).listTemplates({
+                repositoryUuids: `${createdRepos[0].uuid},${createdRepos[1].uuid}`,
+            });
+            expect(templates_repository_uuids_search.data).toHaveLength(4);
+        });
+
+        await test.step('List templates by snapshot UUIDs', async () => {
+            await expect(createdTemplates[0].snapshots?.[0]?.uuid).toBeDefined();
+            const templates_snapshot_uuids_search = await new TemplatesApi(client).listTemplates({
+                snapshotUuids: `${createdTemplates[0].snapshots![0]!.uuid!},${createdTemplates[1].snapshots![0]!.uuid!}`,
+            });
+            expect(templates_snapshot_uuids_search.data).toHaveLength(2);
+        });
+
+        await test.step('List templates by sortBy', async () => {
+            const templates_sort_by_search = await new TemplatesApi(client).listTemplates({
+                sortBy: 'version',
+            });
+            expect(templates_sort_by_search.data?.[0]?.version).toBe('10');
+            expect(templates_sort_by_search.data).toHaveLength(4);
+        });
+
+        await test.step('List templates by limit', async () => {
+            const templates_limit_search = await new TemplatesApi(client).listTemplates({
+                limit: 2,
+            });
+            expect(templates_limit_search.data).toHaveLength(2);
+        });
+
+        await test.step('List templates by offset', async () => {
+            const templates_offset_search = await new TemplatesApi(client).listTemplates({
+                offset: 1,
+                sortBy: 'name',
+            });
+            expect(templates_offset_search.data).toHaveLength(3);
+
+            const sorted_names = createdTemplates
```

Comment:
This could be probably written without map and slice but that is just a detail :)